### DISCUSSION TITLE
Particle boundary conditions

### DISCUSSION THREAD

**Discussion by kli-jfp** [BLAST-WarpX/warpx#4327] [Q&A]
Hello! I'm curious how the particle absorbing boundary condition is implemented. 

1. Does the mirror particle travel into the simulation domain? 

2. Would it be possible to use this mirror-technique without solving Maxwell's eqs. in the guard/mirror region?

3. Can you give me more general information about this implementation?

**Comment by RemiLehe** [comment]
Thanks for the question, and sorry for the late reply!

Just to clarify the question: do I understand correctly that you are asking about the `Reflecting` boundary condition, described in the paragraph below
![Screenshot 2023-10-03 at 4 43 44 PM](https://github.com/ECP-WarpX/WarpX/assets/6685781/4287b4ed-1ed4-42ab-beda-4c0e60861f65)
from the [WarpX documentation](https://warpx.readthedocs.io/en/latest/usage/parameters.html)

If yes, than here are some answers:

1. Yes, the particle is reflected into the simulation domain, and continues to evolve self-consistently in PIC algorithm (i.e. it deposits current/charge and gathers the E/B fields).

2. Could you clarify what you mean by the "guard/mirror" region. In general, in WarpX, the Maxwell equations are solved in all valid cells of the simulation domain. In addition, the overall simulation domain is surrounded by a layer of guard cells, in which boundary conditions are applied according to the user-selected `boundary.field_lo` and `boundary.field_hi` (instead of using the Maxwell equations in these guard cells).

3. The implementation is here: https://github.com/ECP-WarpX/WarpX/blob/development/Source/Particles/ParticleBoundaries_K.H#L67 Let us know if you have specific questions on these.

**Comment by kli-jfp** [comment]
Hey Remi! Thanks for getting back to me. I was referring to the PEC boundary conditions on this page https://warpx.readthedocs.io/en/latest/theory/PML.html. 

"If absorbing boundaries are used, an image charge (equal weight but opposite charge) is considered in the mirror location accross the boundary, and the density from that charge is also deposited in the simulation domain"

I'm guessing this absorbing boundary is used so that charge does not build up at the walls when particles are removed. If I'm understanding the alrogithm correctly, particles that are close the boundary have their opposite charge deposited on the other side of the boundary so that they cancel at the walls. Is this correct?
